content/blog/monitor-application-performance-across-hybrid-cloud-environment-using-apache-skywalking-and-service-mesh.md

```diff
@@ -0,0 +1,331 @@
+````
+
+It should be noted that, in order to avoid collecting ALS logs for the Skupper deployments, e.g., *skupper-router* & *skupper-service-controller* deployed to the same namespace that are used for multicloud communication and hybrid application deployment, we can add the annotation *sidecar.istio.io/inject: false* to those _Skupper_ deployments. After restart deployments, the _Skupper_ deployments will not inject the *istio-proxy* to their pods. 
+
+#### 3 Monitor Online Boutique application
+
+After follow up all the steps, the Online Boutique application metrics will be showing up from Apache SkyWalking UI, under *Service Mesh* tab:
+
+![](/img/sw-app-svc.png)
+
+The SkyWalking UI *Topology* page will show the Online Boutique application topology map: 
+
+![](/img/sw-app-map.png)
+
+You can check the service *Overview* and *Endpoint* pages per service, e.g., *frontend*:
+
+![](/img/sw-app-overview.png)
+
+![](/img/sw-app-endpoint.png)
+
+You can also check all the services from the *Data Plane* tab under  *Service Mesh* that are observed through _Envoy Metrics Service_ setup:
+
+![](/img/sw-app-svc-dataplane.png)
+
+### Online Boutique application alerting
+
+
+
+Apache SkyWalking provides an alerting mechanism to measure application performance according to a list of pre-defined metrics, e.g., *service\_resp\_time*, *service\_instance\_resp\_time*, and *service\_sla*. It will trigger alerting when some metrics reach its pre-defined thresholds. 
+
+A﻿pache SkyWalking configures the alerting using a collection of alerting rules located in */skywalking/config/alarm-settings.yml* from the SkyWalking OAP pod. You can check the content by running the following command:
+
+
+
+```markdown
+$ kukectl exec pod/skywalking-skywalking-helm-oap-bfb57fbf8-5g7k7 -n skywalking -it -- cat /skywalking/config/alarm-settings.yml 
+```
+You can define new metrics by adding new entry to the file using SkyWalking observability analysis language (OAL), or customize the existing metrics with new thresholds. 
+
+
+
+B﻿elow is the alarms page from SkyWalking UI showing all the triggered alerts for deployed Online Boutique application: 
+
+
+
+![](/img/sw-alert.png)
+
+
+
+The alarms page shows an alert * Response time of service instance frontend-549fd9954f-lvnsv of frontend is more than 1000ms in 2 minutes of last 10 minutes *. 
+
+
+
+This alert is triggered by the following metric alerting rule for the metric *service\_instance\_resp\_time*: 
+
+
+
+```markdown
+  service_instance_resp_time_rule:
+    metrics-name: service_instance_resp_time
+    op: ">"
+    threshold: 1000
+    period: 10
+    count: 2
+    silence-period: 5
+message: Response time of service instance {name} is more than 1000ms in 2 minutes of last 10 minutes
+```
+
+
+It indicates issue from the _frontend_ service instance in Online Boutique application. You can check further the service trace page to figure out the root cause for this issue.
```

Comment:
Add the word "an" prior to "issue from"
For the second sentence, I'd modify it to read:
"You can check **the** service trace page **further** to figure out the root cause **of** this issue."